ovh/resource_dbaas_logs_cluster.go

```diff
@@ -0,0 +1,213 @@
+package ovh
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDbaasLogsCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDbaasLogsClusterCreate,
+		Update: resourceDbaasLogsClusterUpdate,
+		Read:   resourceDbaasLogsClusterRead,
+		Delete: resourceDbaasLogsClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDbaasLogsClusterImportState,
+		},
+
+		Schema: resourceDbaasLogsClusterSchema(),
+	}
+}
+
+func resourceDbaasLogsClusterSchema() map[string]*schema.Schema {
+	schema := map[string]*schema.Schema{
+		"service_name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"cluster_type": {
+			Type:        schema.TypeString,
+			Description: "Cluster type",
+			Computed:    true,
+		},
+		"dedicated_input_pem": {
+			Type:        schema.TypeString,
+			Description: "PEM for dedicated inputs",
+			Computed:    true,
+		},
+		"archive_allowed_networks": {
+			Type: schema.TypeSet,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Description: "Allowed networks for ARCHIVE flow type",
+			Optional:    true,
+		},
+		"direct_input_allowed_networks": {
+			Type: schema.TypeSet,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Description: "Allowed networks for DIRECT_INPUT flow type",
+			Optional:    true,
+		},
+		"direct_input_pem": {
+			Type:        schema.TypeString,
+			Description: "PEM for direct inputs",
+			Computed:    true,
+		},
+		"hostname": {
+			Type:        schema.TypeString,
+			Description: "hostname",
+			Computed:    true,
+		},
+		"is_default": {
+			Type:        schema.TypeBool,
+			Description: "All content generated by given service will be placed on this cluster",
+			Computed:    true,
+		},
+		"is_unlocked": {
+			Type:        schema.TypeBool,
+			Description: "Allow given service to perform advanced operations on cluster",
+			Computed:    true,
+		},
+		"query_allowed_networks": {
+			Type: schema.TypeSet,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Description: "Allowed networks for QUERY flow type",
+			Optional:    true,
+		},
+		"region": {
+			Type:        schema.TypeString,
+			Description: "Data center localization",
+			Computed:    true,
+		},
+	}
+
+	return schema
+}
+
+func resourceDbaasLogsClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	serviceName := d.Get("service_name").(string)
+	cluster_id, err := dbaasGetClusterID(config, serviceName)
+	if err != nil {
+		return fmt.Errorf("Error fetching info for %s:\n\t %q", serviceName, err)
+	}
+	d.SetId(cluster_id)
+
+	return resourceDbaasLogsClusterUpdate(d, meta)
+}
+
+func resourceDbaasLogsClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	if false {
```

Comment:
What should i do when releasing the logs cluster from terraform management ?
Should i delete all ACL or should i leave them as-is ?
AFAIK, a empty list mean that the cluster has no ACL

Reply:
so you can't test on computed resource, only the existence of the datasource, and on its service name for example :)

Reply:
When a user execute the command `terraform destroy` on this resource, he/she should have a cluster like it was before the usage of this resource

Reply:
What about this scenario :
- you order a dedicated cluster
- you manage it thru terraform, setting some ACL
- you add data in it
- you stop managing it thru terraform

As you go back to initial state, you don't have ACL anymore and a cluster full of data anyone can query